aws/provider_test.go

```diff
@@ -78,6 +80,44 @@ func testAccPreCheck(t *testing.T) {
 	}
 }
 
+// testAccAwsProviderAccountID returns the account ID of an AWS provider
+func testAccAwsProviderAccountID(provider *schema.Provider) string {
+	if provider == nil {
+		log.Print("[DEBUG] Unable to read account ID from test provider: empty provider")
+		return ""
+	}
+	if provider.Meta() == nil {
+		log.Print("[DEBUG] Unable to read account ID from test provider: unconfigured provider")
+		return ""
+	}
```

Comment:
_minor_ feels like these could end up all over the place, may be worth putting into a central method somewhere that panics for tests?

Reply:
We chatted about this out of band -- currently this will output like this if done incorrectly, which is friendlier than a panic (at least in my opinion):

```
TF_LOG=debug make testacc TEST=./aws TESTARGS='-run=TestAccAWSEcrRepository_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSEcrRepository_basic -timeout 120m
=== RUN   TestAccAWSEcrRepository_basic
2018/08/30 23:05:03 [DEBUG] Unable to read account ID from test provider: empty provider
2018/08/30 23:05:03 [DEBUG] Unable to read account ID from test provider: empty provider
--- FAIL: TestAccAWSEcrRepository_basic (9.20s)
    testing.go:527: Step 0 error: Check failed: Check 4/5 error: aws_ecr_repository.default: Attribute 'registry_id' expected "", got "187416307283"
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	10.447s
```

Luckily we don't have too many `AWSClient` attributes we'll need to fetch off the provider after its initialized, so should just be one or two of these functions added into `provider_test.go` as the current acceptance testing framework is written.

I think we both agree that we should try to fix the ordering for the provider initialization if possible though. The delayed provider configuration in the `PreCheck` is what makes this code worse than it needs to be. I'll take this as a separate, more involved task (likely part of #4610), since for these new functions in `provider_test.go` it should be an implementation detail and should not not fundamentally affect all the resources themselves (hopefully). 👍